src/main/java/pl/damianszczepanik/jenkins/buildhistorymanager/BuildHistoryManager.java

```diff
@@ -49,16 +49,20 @@ public synchronized void perform(Job<?, ?> job) throws IOException, InterruptedE
         // for each completed build...
         while (run != null) {
             LOG.info("Processing build #" + run.getNumber());
-            for (int i = 0; i < rules.size(); i++) {
-                Rule rule = rules.get(i);
-                LOG.info("Processing rule no " + (i + 1));
-                if (rule.validateConditions(run)) {
-                    LOG.info("Processing actions for rule no " + (i + 1));
-                    rule.performActions(run);
+            if (run.isKeepLog()) {
+                LOG.info("build is marked as keep forever -> skipping");
```

Comment:
I need a build number here like below about the rule number - this help debugging potential problems

Reply:
```suggestion
                LOG.info("build #" + run.getNumber() + " is marked as keep forever -> skipping");
```
that would be my next PR after this is merged, because only the build number will not help if you configured this for multiple builds the build name is also needed!